### PR TITLE
added more tests to neils PR

### PR DESCRIPTION
**Pull request by GeorgeJahad**
Just added a couple of tests cases to TestUgiFilter.  (bad date, and bad aws version.)

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java [hunk 1]
```diff
@@ -131,19 +127,57 @@ public void testUgiFilterDoFilter() throws IOException, ServletException {
...
+      filter.doFilter(request, response, filterChain);
+      filter.destroy();
+      Assert.fail("Filter should generate OS3 exception.");
+    } catch(Exception e) {
+      Assert.assertTrue(e.getCause().getCause() instanceof OS3Exception);
+    }
+
+  }
+
+  private void resetAuthHeader(HttpServletRequest request,
```
Comment (neils-dev) [inline hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java]: We can do away with this function to reset the Auth header for the different test cases.  Here we have 3 - 
i.) fail due to incorrect date in auth header. 
ii.) pass by correcting date to curdate in auth header
iii.) fail due to unsupported version
In each of these cases it affects (through the UgiFilter calls) the `getHeaderNames()` and `getHeader` with AUTHORIZATION_HEADER.  Each invocation of these methods by the UgiFilter (called for each of the 3 test cases) needs to return a new enumeration collection header names and an updated header AUTHORIZATION_HEADER respectively.  
Instead of creating new mock methods through the `resetAuthHeader`, we can have the `Mockito.when` return new values per invocation (UgiFilter calls to the `getHeaderNames` and `getHeader` with updated AUTHORIZATION_HEADER).  I suggest we change the Mockito methods for `getHeaderNames` and `getHeader` for auth to:

```
    Mockito.when(request.getHeaderNames()).thenReturn(
        Collections.enumeration(headers.keySet()),
        Collections.enumeration(headers.keySet()),
        Collections.enumeration(headers.keySet()));

```
```
    Mockito.when(request.getHeader(AUTHORIZATION_HEADER)).thenAnswer(new Answer() {
      public String answer(InvocationOnMock invocation) throws Throwable {
        return headers.get(AUTHORIZATION_HEADER);
      }
    });

```

With that the values returned by the mocked methods return new headerNames and updated header auth with each of the three i-iii test cases.

line 141 can now be changed to just updating the header auth :
```
    headers.put(AUTHORIZATION_HEADER, headers.
        get(AUTHORIZATION_HEADER).replace("20210616", curDate));
    try {

```
and line 155 can be:
```
    headers.put(AUTHORIZATION_HEADER, headers.
        get(AUTHORIZATION_HEADER).replace("AWS4", "AWS3"));
    try {
```

File: hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java [hunk 2]
```diff
@@ -131,19 +127,57 @@ public void testUgiFilterDoFilter() throws IOException, ServletException {
         headers.get(LENGTH_HEADER));
     Mockito.when(request.getParameterMap()).thenReturn(parameters);
 
+    // Should generate exception because of incorrect date
```
Comment (neils-dev) [inline hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java:126]: We can also add a test case for testing the `validateAccessId` in the` UgiFilter`, have an empty user string.

**Review by neils-dev — CHANGES_REQUESTED** [review]

File: hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java (hunk 1 above)
Comment (neils-dev) [reply hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java]: looks good.

File: hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java (hunk 2 above)
Comment (neils-dev) [reply hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java:126]: Can add test for validateAccessId -> checked in UgiFilter doFilter.  Ok.  Missed it.  - Thanks

Comment (neils-dev) [reply hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java:126]: Saw it at end.  Looks good.
